docs/contributing/guides/stacks/crash-course/rust-crash-course.md

```diff
@@ -0,0 +1,357 @@
+# Arrow Rust Crash Course
+
+Through these projects you will learn what you need to contribute to the Arrow Rust codebase.
+
+Attempt these steps without looking them up first! You'll only learn by trying them yourself.
+
+You're meant to struggle a bit - don't worry! The compiler will help :)
+
+When you're ready for help, many of the instructions link to helpful resources or Rust docs.
+
+## Level 0 - Libraries, Tests, Refs
+
+1) Compile the [Hello, Cargo!](https://doc.rust-lang.org/book/ch01-03-hello-cargo.html) project
+
+2) [Create src/lib.rs and add it as a library to Cargo.toml](https://doc.rust-lang.org/cargo/reference/cargo-targets.html?highlight=library#configuring-a-target)
+
+<details><summary>Solution</summary>
+<p>
+
+```toml
+# in Cargo.toml
+[lib]
+name = "adder"
+path = "src/lib.rs"
+test = true
+```
+</p>
+</details>
+
+3) Add a function to `src/lib.rs` which takes an unsigned integer as an argument and [returns](https://doc.rust-lang.org/rust-by-example/fn.html) the sum.
+
+```rust
+fn add_one(x: u32) -> u32 {
+    // what are two ways to return a value from this function?
+    // ...
```

Comment:
Is there a solution for this one?

Reply:
I can add it

Reply:
Added https://github.com/Arrow-air/website/pull/62/commits/3c08e0b7ad20ed0e8d49547256f67aba628fb3fd